mars/dataframe/base/reset_index.py

```diff
@@ -0,0 +1,197 @@
+        in_df = op.inputs[0]
+        out_df = op.outputs[0]
+        added_columns_num = len(out_df.dtypes) - len(in_df.dtypes)
+        out_chunks = []
+        range_index = isinstance(out_df.index_value.value, IndexValue.RangeIndex)
+        cum_range = np.cumsum((0, ) + in_df.nsplits[0])
+        for c in in_df.chunks:
+            if range_index:
+                index_value = parse_index(pd.RangeIndex(cum_range[c.index[0]], cum_range[c.index[0] + 1]))
+            else:
+                index_value = out_df.index_value
+            if c.index[1] == 0:
+                chunk_op = op.copy().reset_key()
+                dtypes = out_df.dtypes[:(added_columns_num + len(c.dtypes))]
+                columns_value = parse_index(dtypes.index)
+                new_chunk = chunk_op.new_chunk([c], shape=(c.shape[0], c.shape[1] + added_columns_num),
+                                               index=c.index, index_value=index_value,
+                                               columns_value=columns_value, dtypes=dtypes)
+            else:
+                chunk_op = op.copy().reset_key()
+                chunk_op._drop = True
+                new_chunk = chunk_op.new_chunk([c], shape=c.shape, index_value=index_value,
+                                               index=c.index, columns_value=c.columns_value, dtypes=c.dtypes)
+            out_chunks.append(new_chunk)
+        new_op = op.copy()
+        columns_splits = list(in_df.nsplits[1])
+        columns_splits[0] += added_columns_num
+        nsplits = (in_df.nsplits[0], tuple(columns_splits))
+        return new_op.new_dataframes(op.inputs, out_df.shape, nsplits=nsplits,
+                                     chunks=out_chunks, dtypes=out_df.dtypes,
+                                     index_value=out_df.index_value, columns_value=out_df.columns_value)
+
+    @classmethod
+    def tile(cls, op):
+        if isinstance(op.inputs[0], DATAFRAME_TYPE):
+            return cls._tile_dataframe(op)
+        else:
+            return cls._tile_series(op)
+
+    @classmethod
+    def execute(cls, ctx, op):
+        in_data = ctx[op.inputs[0].key]
+        out = op.outputs[0]
+
+        kwargs = dict()
+        if op.name is not None:
+            kwargs['name'] = op.name
+        if op.col_level is not None:
+            kwargs['col_level'] = op.col_level
+        if op.col_fill is not None:
+            kwargs['col_fill'] = op.col_fill
+
+        r = in_data.reset_index(level=op.level, drop=op.drop, **kwargs)
+        if isinstance(out.index_value.value, IndexValue.RangeIndex):
+            r.index = out.index_value.to_pandas()
+        ctx[out.key] = r
+
+    def _call_series(self, a):
+        if self.drop:
+            index_value = parse_index(pd.RangeIndex(a.shape[0]))
+            return self.new_series([a], shape=a.shape, dtype=a.dtype, name=a.name, index_value=index_value)
+        else:
+            empty_df = pd.Series(index=a.index_value.to_pandas(), dtype=a.dtype, name=a.name)
```

Comment:
Rename `empty_df` to `empty_series` and make a utils for `build_empty_series` just like `build_empty_df`?